models/task.rb

```diff
@@ -0,0 +1,136 @@
+require_relative '../table_setup.rb'
+require_relative '../services/database_connection.rb'
+require_relative 'user.rb'
+
+class Task
+  @@connection = nil
+
+  def self.connection
+    @@connection ||= DatabaseConnection.connection 
+  end
+
+  attr_accessor :assignee_user_id, :description, :due_date, :priority, :creator_id, :status, :id
+
+  def initialize(assignee_user_id:, description:, due_date:, priority:, creator_id:, status:, id: nil)
+    self.id = id
+    self.assignee_user_id = assignee_user_id
+    self.description = description
+    self.due_date = due_date
+    self.priority = priority
+    self.creator_id = creator_id
+    self.status = status
+  end
+
+  def save 
+    begin
+      Task.connection.exec_params("INSERT INTO tasks(assignee_user_id, description, due_date, priority, creator_id, status) VALUES($1, $2, $3, $4, $5, $6)", [assignee_user_id, description, due_date, priority, creator_id, status])
+    rescue PG::InvalidTextRepresentation => e   
+      puts "Invalid Represetation (May be status wrong)"
+      puts e.message
+    rescue PG::SyntaxError => e
+      puts 'Error: A syntax error occurred in the SQL query.'
+      puts e.message
+    rescue PG::ConnectionBad => e
+      puts 'Error: The database connection is invalid or closed.'
+      puts e.message 
+    else
+      puts "Task Assign Successfully"
+    end
+  end
+
+  def self.find_task(task_id)
+    res = @@connection.exec_params("SELECT * FROM tasks WHERE id=$1", [task_id])
+    if res.ntuples >= 1
```

Comment:
if searching by id, we should find exactly one entry. This if comparison is confusing. It should never return two entries. 
We should rather write like:
```ruby
if res.ntuples.zero?
  # error
else
  row = result[0] # please rename "res" to "result". Don't use short forms that indicate half the meaning
  new(**result.transform_keys(&:to_sym)) # this is a shortcut, read about it
end
```